pkgs/os-specific/linux/nvidia-x11/builder.sh

```diff
@@ -73,8 +73,14 @@ installPhase() {
             else
                 sed -E "s#(libGLX_nvidia)#$i/lib/\\1#" nvidia_icd.json > nvidia_icd.json.fixed
             fi
-            install -Dm644 nvidia_icd.json.fixed $i/share/vulkan/icd.d/nvidia.json
+
+            if [ "$system" = "x86_64-linux" ]; then
+                install -Dm644 nvidia_icd.json.fixed $i/share/vulkan/icd.d/nvidia_icd.x86_64.json
+            elif [ "$system" = "i686-linux" ]; then
```

Comment:
Unfortunately, this only works with the legacy driver versions. In the newer ones, the 32bit libs are bundeled in the x86_64 package, so `$system` will always be x86_64
```
$ nix-build -A linuxPackages.nvidia_x11.lib32
$ ls result-lib32/share/vulkan/icd.d
nvidia_icd.x86_64.json
```
See also https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/hardware/video/nvidia.nix#L28-L32

Reply:
Updated as per https://github.com/NixOS/nixpkgs/issues/86620#issuecomment-624180239

Though if it isn't ideal, I'll change it again.

The conditional scheme should work fine, since the only way to get the non-bundled version is from building from i686 platform I believe. Though using `i686bundled` might be preferred?